posthog/models/insight.py

```diff
@@ -120,3 +120,15 @@ def dashboard_item_saved(sender, instance: Insight, dashboard=None, **kwargs):
         filter = get_filter(data=instance.dashboard_filters(dashboard=dashboard), team=instance.team)
 
         instance.filters_hash = generate_cache_key("{}_{}".format(filter.toJSON(), instance.team_id))
+
+
+class InsightViewed(models.Model):
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(fields=["team", "user", "insight"], name="posthog_unique_insightviewed"),
```

Comment:
Good call. Just added an index on `["team_id", "user_id", "-last_viewed_at"]`. I'm a bit hesitant to add an index on `last_viewed_at` alone because it's one query once a day that doesn't really have to be that performant. But I don't feel that strongly.